kyo-prelude/shared/src/main/scala/kyo/Parse.scala

```diff
@@ -28,9 +37,9 @@ object Parse:
       * @return
       *   Parsed value if successful, drops the current parse branch if unsuccessful
       */
-    def read[A, S](f: Text => Maybe[(Text, A)] < S)(using Frame): A < (Parse & S) =
+    def read[A](f: Text => Maybe[(Text, A)])(using Frame): A < Parse =
         Var.use[State] { state =>
-            f(state.remaining).map {
+            readAspect(state.remaining)(f).map {
```

Comment:
The aspect is applied only to the read function. It enables users to customize the reading behavior for an entire computation like in the new `Parse.spaced`.